### PR TITLE
MAINT: match signature between corner_peaks and peak_local_max

### PR DESCRIPTION
**Pull request by sciunto**
## Description

It seems that after some changes in peak_local_max, corner_peaks has not been updated  (changes like https://github.com/scikit-image/scikit-image/commit/f3771aba7e097f769bd5b538a8347c9d40c1b7e5#diff-30c3e44d26344e4c11a27bdd1ae43731).

This PR attempts to fix this. @jni, the body guard of API changes can let me know if it is OK :)

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by sciunto** [comment]
@scikit-image/core Anyone to review this PR? Or should we target this for 0.17?

File: skimage/feature/corner.py [hunk 1]
```diff
@@ -950,13 +950,20 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=0.1,
     array([[2, 2]])
 
     """
+    if threshold_rel is None:
+        threshold_rel = 0.1
+        warn("Until the version 0.16, threshold_rel was set to 0.1 by default."
+             "Starting from version 0.16, the default value is set to None."
+             "Until version 0.18, a None value corresponds to a threshold value of 0.1."
+             "The default behavior will match skimage.feature.peak_local_max.")
```
Comment (jni) [inline skimage/feature/corner.py]: I think this should say what the `peak_local_max` default actually is...

Comment (lagru) [reply skimage/feature/corner.py]: It defaults to `max(threshold_abs, threshold_rel * image.max())` and `threshold_abs` defaults to `image.min()`. The connection between `threshold_abs` and `threshold_rel` is not documented in `peak_local_max`. Might be better to explain it there and just pointed the warning at that explanation considering that it's not trivial and this function's docstring already does the same.

Comment (lagru) [reply skimage/feature/corner.py]: @sciunto could you add `stacklevel=2` and `category=FutureWarning` here?

**Comment by lagru** [comment]
I left one suggestion but looks good otherwise.

File: skimage/feature/corner.py (hunk 1 above)
Comment (sciunto) [reply skimage/feature/corner.py]: I'm OK with the idea to improve docstrings, but i would like to leave it for another PR.

File: TODO.txt [hunk 2]
```diff
@@ -38,6 +38,14 @@ Version 0.18
...
   to 2.
 
+Version 0.18
+------------
+* In ``skimage.features.corner_peaks``, remove the warning.
+
+Version 0.18
+------------
+* In ``skimage.features.corner_peaks``, remove the warning.
+
```
Comment (rfezzani) [inline TODO.txt]: This is duplicated... More over, the section title `Version 0.18` is not necessary, simply add the item to the list.

Comment (sciunto) [reply TODO.txt]: It's an issue with the merge. I'll fix it later if noone does.

**Comment by jni** [comment]
@sciunto fwiw I am guardian of the dimensions, not the API, though I tend to have strong opinions about the API. =)

**Comment by sciunto** [comment]
:)

**Pull request merged**